### DISCUSSION TITLE
CI workflow (on pull_request and event opened) is not getting triggered when a new PR is opened by a dispatch workflow

### DISCUSSION THREAD

**Discussion by radhikari-arch** [community/community#25251] [Actions]
<p>Hi,<br>
I have this case where a status check set was on a branch policy which is a CI that I am expecting to run when PR is opened. Well, this works when I open the PR manually, but not when PR was opened by a dispatch workflow.<br>
I have a dispatch workflow create_pr.yml that creates a PR with some inputs:<br>
create_pr.yml</p>
<pre><code class="lang-auto">name: Create PR

on:
  workflow_dispatch:
      inputs:
       ......

  jobs:
  create_pr:
    name: Create PR test
    runs-on: [self-hosted, dev]
    
    steps:
      .......
      - name: Create a working branch and push repo
        run: |
          git pull &amp;&amp; git checkout -b automation/ticket1234
          ## Add/change files
          .....
          git add -A &amp;&amp;
          git commit -m "Github Automated Workflow - deployment file added" &amp;&amp;
          git push -u origin automation/ticket1234

       - name: Create a Pull Request
          uses: actions/github-script@v6
        with:
          script: |
            const result = await github.rest.pulls.create({
              title: 'Create resource',
              owner: context.repo.owner,
              repo: context.repo.repo,
              head: 'automation/ticket1243',
              base: 'development',
              body: [
                  'test body',
                  'line 2'
              ].join('\n')
            });
</code></pre>
<p>And I have a CI workflow. ci.yml, that runs on following different events of pull requests. I have this CI added into the status check for the “development” branch policy. All other pull request events works fine once the PR is opened that triggers the ci.yml workflow and sends the status check to the PR. However, the status check keeps on waiting when the PR is first open and the workflow does not get triggered.<br>
ci.yml</p>
<pre><code class="lang-auto">name: CI - Check terraform plan

on:
  pull_request:
    types: [opened, reopened, synchronize, ready_for_review, labeled, review_requested, edited]
    branches:
      - 'development'
    paths:
      - '**.tf'
...
</code></pre>
<p>I tried the workflow_call on the CI and called it from the create_pr.yml workflow, howevere the context of the opened PR is not passed. So that is not going to work for me. So how can I make the CI run as soon as the PR was opened. Only workaround I have now is that CR gets triggered whenever I changed other events on the PR after it is created, like adding labels, edit, add reviewers or put in and out of draft mode.</p>

**Comment by radhikari-arch** [comment]
<p>This is what it looks like when a PR is opened by workflow dispatch, and CI won’t run until I change other events, like add a label or put in and out of draft etc. I am looking for a way so that CI ci.yml is run as soon as the PR was created by dispatch workflow create_pr,yml.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/c/cc891e00fdfca7a0f9d3770ff99392cda97f70ed.png" data-download-href="/uploads/short-url/tbpaVoY83IcxCjvCryiQQQHLTGJ.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/70524094/181089862-b38be861-357e-4f6f-851d-ec821ac73eec.png" alt="image" data-base62-sha1="tbpaVoY83IcxCjvCryiQQQHLTGJ" width="690" height="207" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/c/cc891e00fdfca7a0f9d3770ff99392cda97f70ed_2_690x207.png, https://aws1.discourse-cdn.com/github/original/3X/c/c/cc891e00fdfca7a0f9d3770ff99392cda97f70ed.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/c/c/cc891e00fdfca7a0f9d3770ff99392cda97f70ed.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/c/cc891e00fdfca7a0f9d3770ff99392cda97f70ed_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">947×285 20 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by jsoref** [comment]
<aside class="quote" data-post="2" data-topic="238439">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/githhub-acitons-push-event-not-triggered-if-githubactions-bot-enabled-auto-merge/238439/2">GithHub Acitons: Push Event not triggered if githubactions[bot] enabled Auto merge</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    Correct. This is by <a href="https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow">design</a>
  </blockquote>
</aside>

<p>If you want to trigger something, you’ll need to use a:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token">
  <header class="source">
      <img src="https://docs.github.com/assets/cb-803/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token" target="_blank" rel="noopener">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/2119212/181089863-43dac9e5-8aec-4daf-9c5f-4690cf86bd35.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token" target="_blank" rel="noopener">Creating a personal access token - GitHub Docs</a></h3>

  <p>You should create a personal access token to use in place of a password with the command line or with the API.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Reply by InDieTasten** [reply]
The page link you provided as "by design" does not indicate anything about GITHUB_TOKEN being unable to trigger other workflows, and subsequently also doesn't make any statements that would tell us whether it's by design or not.

**Comment by radhikari-arch** [comment]
<p>Yes, I had that PAT used, I had it redacted from the original post. Here is the create_pr.yml and ci.yml again. Expectation here is ci.yml to run as soon as create_pr.yml completes that creates a new PR. ci.yml is on the branch check status and shows “waiting” and only runs until other events are triggered on the PR manually.<br>
Do I need to pass any permissions as showsn here: <a href="https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow" class="inline-onebox" rel="noopener nofollow ugc">Automatic token authentication - GitHub Docs</a><br>
If so what exact permissions are needed.</p>
<p>create_pr.yml</p>
<pre><code class="lang-auto">name: Create PR

on:
  workflow_dispatch:
      inputs:
       ......

  jobs:
  create_pr:
    name: Create PR test
    runs-on: [self-hosted, dev]
    
    steps:
      .......
      - name: Setup Git Config
        run:
          git config --global url."https://oauth2:${GH_TOKEN}@github.com/MY-ORG".insteadOf "https://github.com/MY-ORG"
        env:
          GH_TOKEN: "${{ secrets.MY_PAT }}"

      - name: repo checkout
        uses: actions/checkout@v2
        with:
          token: "${{ secrets.MY_PAT }}"
          ref: 'development'

      - name: Create a working branch and push repo
        run: |
          git pull &amp;&amp; git checkout -b automation/ticket1234
          ## Add/change files
          .....
          git add -A &amp;&amp;
          git commit -m "Github Automated Workflow - deployment file added" &amp;&amp;
          git push -u origin automation/ticket1234

       - name: Create a Pull Request
          uses: actions/github-script@v6
        with:
          script: |
            const result = await github.rest.pulls.create({
              title: 'Create resource',
              owner: context.repo.owner,
              repo: context.repo.repo,
              head: 'automation/ticket1243',
              base: 'development',
              body: [
                  'test body',
                  'line 2'
              ].join('\n')
            });
			
</code></pre>
<p>And, ci.yml</p>
<pre><code class="lang-auto">name: CI - Check terraform plan

on:
  pull_request:
    types: [opened, reopened, synchronize, ready_for_review, labeled, review_requested, edited]
    branches:
      - 'development'
    paths:
      - '**.tf'
...
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<p>You don’t need the PAT for the Git operations at all, you need it when creating the PR. And the <code>actions/github-script</code> step doesn’t see the token stored in the Git configuration, you need to provide it as described in the documentation:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/actions/github-script#using-a-separate-github-token">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/actions/github-script#using-a-separate-github-token" target="_blank" rel="noopener">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/5814919/181089865-06d68f28-c495-4dc7-b42d-87a399aadf60.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/5/7/572e8e0197667d4d891eeb1f311689ef89be6ecf_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/5/7/572e8e0197667d4d891eeb1f311689ef89be6ecf_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/5/7/572e8e0197667d4d891eeb1f311689ef89be6ecf.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/5/7/572e8e0197667d4d891eeb1f311689ef89be6ecf_2_10x10.png"></div>

<h3><a href="https://github.com/actions/github-script#using-a-separate-github-token" target="_blank" rel="noopener">GitHub - actions/github-script: Write workflows scripting the GitHub API in...</a></h3>

  <p>Write workflows scripting the GitHub API in JavaScript - GitHub - actions/github-script: Write workflows scripting the GitHub API in JavaScript</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>I recommend you <em>remove</em> the PAT from the Git config to minimize available privileges. The <code>GITHUB_TOKEN</code> should be sufficient there, unless you also want to trigger workflows on push or something.</p>

**Reply by InDieTasten** [reply]
I don't get why this was marked as the answer. The initial question was exactly about why "on: pr" is not triggered, when a PR is created by "github-actions (bot)", aka the GITHUB_TOKEN.

OP is able to create the PR. The issue is, he cannot merge it without build validation workflows being initiated

**Comment by radhikari-arch** [comment]
<p>Thanks, this resolved my issue. I removed the token from create_pr.yml workflow and then passed MY_PAT only at the actions/ github-scripts step. That executed the CI when the pr got opened.</p>

**Reply by thalysonalexr** [reply]
what permissions did you pass to the token?

**Comment by jsoref** [comment]
<p>Do note that <code>PAT</code>s should be stored in secrets – not committed into the repository.</p>

**Comment by InDieTasten** [comment]
The following place in the docs explain this behavior:

> When you use the repository's `GITHUB_TOKEN` to perform tasks, events triggered by the `GITHUB_TOKEN`, with the exception of `workflow_dispatch` and `repository_dispatch`, will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs. For example, if a workflow run pushes code using the repository's `GITHUB_TOKEN`, a new workflow will not run even when the repository contains a workflow configured to run when `push` events occur.
> 
> Commits pushed by a GitHub Actions workflow that uses the `GITHUB_TOKEN` do not trigger a GitHub Pages build.

*Source: https://docs.github.com/en/actions/concepts/security/github_token#when-github_token-triggers-workflow-runs*
